Úprava autocomplete, nový seznam PSČ, oprava validátoru PSČ

KamilZm opened the pull request:
- úprava autocomplete aby nabízel hodnoty zleva (upraven zdroják komponenty, vuejs-auto-complete-0.9.0-patched-min.js)
- oprava validátoru PSČ (PSČ nemůže začínat na 8 nebo 9 + tam byl ještě starý formát s mezerou)
- přidán assets\cityzipname.js se seznamem (snad) všech PSČ čistě pro našeptávač
- úprava vyplňování údajů o obci (zadané PSČ se snaží dohledat původním seznamu assets\citycouncils.js)

KamilZm committed:
Merge pull request #1 from HlidacStatu/euro2019

Merge z  HlidacStatu

KamilZm committed:
Merge pull request #2 from HlidacStatu/euro2019

Merge z Hlídače

KamilZm committed:
Úprava autocomplete aby nabízel hodnoty zleva
Oprava validátoru PSČ (PSČ nemůže začínat na 8 nebo 9 + tam byl ještě starý formát s mezerou)

KamilZm committed:
- přidán assets\cityzipname.js se seznamem (snad) všech PSČ čistě pro našeptávač
- úprava vyplňování údajů o obci (zadané PSČ se snaží dohledat původním seznamu assets\citycouncils.js)

zzen commented on index.html:
```diff
@@ -31,21 +31,22 @@
+    <script src="assets/vuejs-auto-complete-0.9.0-patched-min.js"></script>
     <script src="assets/lang-cs.js"></script>
-    <script src="assets/citycouncils.min.js"></script>
+    <script src="assets/citycouncils.min.js"></script>  <!-- seznam obecnich uradu, ale nejsou tam vsechny -->
```
Měli jsme před pár týdny v č.d Slacku hlasování a shodli jsme se, že zdrojáky včetně komentářů chceme držet v angličitně - mohu poprosit o úpravy těch komentářů? Když už tam budete dělat změny, šlo by tam také napsat, proč je dobré mít ty seznamy dva (tj. že na citycouncils.js momentálně nemáme úplný dataset a tak jsme udělali úplný aspoň seznam PSČ?)

zzen commented on index.html:
```diff
@@ -31,21 +31,22 @@
     <script>
         window.dataLayer = window.dataLayer || [];
-        function gtag(){dataLayer.push(arguments);}
+        function gtag() { dataLayer.push(arguments); }
```
tady poprosím zbytečně neměnit historii editace řádků jen kvůli whitespace

KamilZm replied:
Tohle mi bohužel dělá Visual Studio. Snažím se to vracet, ale někdy to uteče.

zzen commented on index.html:
```diff
@@ -31,21 +31,22 @@
     <script src="assets/lang-cs.js"></script>
-    <script src="assets/citycouncils.min.js"></script>
+    <script src="assets/citycouncils.min.js"></script>  <!-- seznam obecnich uradu, ale nejsou tam vsechny -->
+    <script src="assets/cityzipname.min.js"></script>   <!-- seznam psc + obci (snad vsech) pro naseptavac -->
```
Dobrý nápad! Potom ale můžeme vyhodit `zip_name` klíč z `citycouncils(.min).js` a ušetřit trochu místa na lince.

KamilZm replied:
OK

zzen commented on index.html:
```diff
@@ -753,20 +755,34 @@ <h2>3. We wish you good voting!</h2>
-                    this.userData.municipality.databox = result.selectedObject.datovaSchranka;
-                    this.userData.municipality.email = result.selectedObject.email;
+
+                    // pokusim se najit podle PSC dalsi udaje v cities
```
komentáře v angličtině prosím

KamilZm replied:
Budu se snažit, ale má Angličtina ....

zzen commented on index.html:
```diff
@@ -753,20 +755,34 @@ <h2>3. We wish you good voting!</h2>
+                        this.userData.municipality.databox = selectedCity.datovaSchranka;
+                        this.userData.municipality.email = selectedCity.email;
+                    }
+                    else {  // nenalezeno - radeji udaje ve formulari smazu, aby nebyly spatne (napr. z minuleho hledani)
```
komentáře v angličtině prosím

zzen reviewed (changes requested):
díky za pullrequest, prosím o drobné úpravy

KamilZm committed:
- přechod na nový seznam úřadů (citycouncils_alt.js, původní citycouncils.js zatím ponechán v repo), dohledávají se před kód obce/městské části (ještě to ale není konečné řešení s adresnímí místy)
- doděláno tlačítko na smazání obsahu formuláře
- doplnění valudátorů
- další drobné úpravy (mírné sjednocení názvů atd.)
